### PR TITLE
enh: store awareness messages separately

### PR DESCRIPTION
**Pull request by max-nextcloud**
### 📝 Summary

* Resolves: #3732 

### 🚧 TODO

- [x] fix migration

### 🏁 Checklist

- [x] Code is properly formatted (`npm run lint` / `npm run stylelint` / `composer run cs:check`)
- [x] [Sign-off message](https://probot.github.io/apps/dco/) is added to all commits
- [ ] [Tests](https://github.com/nextcloud/text#-testing-the-app) (unit, integration and/or end-to-end) passing and the changes are covered with tests
- [x] Documentation ([README](https://github.com/nextcloud/text/blob/main/README.md) or [documentation](https://github.com/nextcloud/documentation/blob/master/admin_manual/configuration_server/text_configuration.rst#L2)) has been updated or is not required

### CODE REVIEW

File: src/services/WebSocketPolyfill.js [hunk 1]
```diff
@@ -88,19 +88,33 @@ export default function initWebSocketPolyfill(syncService, fileId, initialSessio
 		}
 
 		#initiateSending() {
-			let steps
+			let outbox
 			syncService.sendSteps(() => {
-				steps = this.#queue.map(s => encodeArrayBuffer(s))
+				outbox = this.#queue
```
Comment (juliushaertl) [inline src/services/WebSocketPolyfill.js:93]: When testing it seems that debouncing needs to be improved now for the push request, it seems on my test with two browsers the push is triggering every 200ms through the following stack trace even if the cursor positions do not change:

Screenshot 2023-02-08 at 12 10 59

File: lib/Service/ApiService.php [hunk 2]
```diff
@@ -184,11 +184,12 @@ public function close($documentId, $sessionId, $sessionToken): DataResponse {
 	 * @throws NotFoundException
 	 * @throws DoesNotExistException
 	 */
-	public function push($documentId, $sessionId, $sessionToken, $version, $steps, $token = null): DataResponse {
+	public function push($documentId, $sessionId, $sessionToken, $version, $steps, $awareness, $token = null): DataResponse {
 		if (!$this->sessionService->isValidSession($documentId, $sessionId, $sessionToken)) {
 			return new DataResponse([], 403);
 		}
 		$session = $this->sessionService->getSession($documentId, $sessionId, $sessionToken);
+		$this->sessionService->updateSessionAwareness($documentId, $sessionId, $sessionToken, $awareness);
```
Comment (juliushaertl) [inline lib/Service/ApiService.php:192]: Would be good to return early here if no steps are passed along.

File: src/services/WebSocketPolyfill.js (hunk 1 above)
Comment (max-nextcloud) [reply src/services/WebSocketPolyfill.js:93]: I suspect this is due to us replaying old awareness messages and creating an inconstent state.
I created #3785 to track it.

**Review by juliushaertl — APPROVED** [review]
👍 

Also tested to collaborate a bit with @miaulalala and no collaboration issues encountered 🚀

**Pull request merged**